client/src/components/Interface.js

```diff
@@ -0,0 +1,238 @@
+import React from 'react';
+import {withRouter} from 'react-router-dom';
+import {connect} from 'react-redux';
+import {initialize, move, say, shout} from '../actions/index';
+import Pusher from 'pusher-js';
+
+class Interface extends React.Component {
+
+    componentWillMount = () =>{
+        if(localStorage.getItem('jwt')){
+            this.gameInit()
+        } else {
+            this.props.history.push('/login')
+        }
+    }
+
+    componentDidMount = () =>{
+        if(localStorage.getItem('uuid')){
+            this.pusherSubscribe();
+        }
+    }
+
+    componentWillReceiveProps(newProps){
+        if(newProps.readout !== this.state.readout){
+            let newPaths = this.state.paths;
+            newPaths.unshift(newProps.readout);
+            this.setState({
+                readout: newProps.readout,
+                paths: newPaths 
+            })
+        }
+    }
+
+    componentWillUnmount(){
+        if(this.channel){
+            this.channel.unbind();
+            this.pusher.unsubscribe(this.channel);
+        }
+    }
+
+    constructor(props){
+        super(props);
+        this.state = {
+            readout : this.props.readout,
+            paths: [],
+            command : '',
+            events: []
+        }
+    }
+
+    updateEvents = data => {
+        let newArray = this.state.events.slice(0);
+        newArray.unshift(data);
+        // window.alert(data.message)
+        this.setState({
+            events: newArray,
+        })
+    }
+
+    handleInput = event => {
+        event.preventDefault();
+        this.setState({
+            [event.target.name]: event.target.value
+        })
+    }
+
+    gameInit = () =>{
+        var token = localStorage.getItem('jwt');
+        this.props.initialize(token);
+    }
+
+    pusherSubscribe = () =>{
+        let uuid = localStorage.getItem('uuid');
+        let pusherKey = process.env.REACT_APP_PUSHER_KEY;
+        let pusherCluster = process.env.REACT_APP_PUSHER_CLUSTER;
+        this.pusher = new Pusher(pusherKey, {
+            cluster: pusherCluster,
+            encrypted: true,
+        })
+         
+        // subscribes to the current user's channel
+        // local room /say will post to this channel, as well as global /shout and private /whisper (not yet functional)
+        this.channel = this.pusher.subscribe(`p-channel-${uuid}`);
+        this.channel.bind('broadcast', this.updateEvents);
+    }
+
+    handleSubmit = event => {
+        event.preventDefault();
+        // store token for command inputs to server
+        let token = localStorage.getItem('jwt');
+
+        // parse movements
+        // if(this.state.command === 'n' || this.state.command === 's' || this.state.command === 'e' || this.state.command === 'w'){
+        if(this.state.command && this.state.command[0] !== '/'){
+            let direction = '';
+            switch(this.state.command){
```

Comment:
This is good but looking at the syntax it makes me wonder if you couldn't have just set the direction variable directly from the command attribute instead of using a switch statement to set it (and then having error handling if the user inputs an invalid command). Maybe I'm not seeing the full picture of this method though.

Reply:
I think this was more of an exercise in utilizing switch/case statements than anything else, as I haven't built them as often as I've built nested if statements, which can become jumbled and messy if they go on too long. The idea behind this input sanitization was to ensure that the backend receives the proper value, which is just a single character direction, instead of the full direction string (e.g. 'n' vs 'north'), while still giving users some freedom as to how they want to move directions.